stock_picking_partner_note/models/res_partner.py

```diff
@@ -12,3 +12,15 @@ class ResPartner(models.Model):
         comodel_name="stock.picking.note",
         string="Picking Notes",
     )
+
+    def write(self, vals):
+        # Update notes on open pickings
+        res = super().write(vals)
+        if "stock_picking_note_ids" in vals:
+            self.env["stock.picking"].search(
+                [
+                    ("partner_id", "child_of", self.ids),
+                    ("state", "in", ("draft", "waiting", "confirmed", "assigned")),
+                ]
```

Comment:
never modify a printed picking
```suggestion
                    ("printed", "=", False),
                ]
```